lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -1584,6 +1701,8 @@ def has_different_configuration(self, image):
             volumes_from=host_config.get('VolumesFrom'),
             working_dir=config.get('WorkingDir'),
             publish_all_ports=host_config.get('PublishAllPorts'),
+            expected_healthcheck=config.get('Healthcheck'),
+            disable_healthcheck=(not config.get('Healthcheck') or config.get('Healthcheck').get('Test') == ['NONE']),
```

Comment:
I think the `disable_healthcheck` is redundant, we can leave the logic to `comparisons` and users to handle it. Introducing this logic might also conflict with `docker`'s original logic, as `docker run --health-cmd ANYTHING` will always generate a "CMD-SHELL" entry in `HealthCheck.Test`(tested on CentOS with docker 1.31).

Reply:
It doesn't really work that way, since docker indicates "no healthcheck" by returning `None` (I think), which isn't compatible to what `comparisons` does.

I don't understand why this logic (you mean `disable_healthcheck`?) conflicts with `docker`'s original logic?

In any way, I think we have to convert the string syntax for `test` to `CMD-SHELL`, otherwise idempotency won't work. That's a good point! (We should also have a test for it.)

Reply:
Yes, so I think we just need to `strict` compare these two options.

Reply:
> In any way, I think we have to convert the string syntax for test to CMD-SHELL

Right. There could be easy fix, like:
```
if isinstance(result[key], (tuple, list)):
    result[key] = [str(e) for e in result[key]]
else:
    result[key] = ["CMD-SHELL", str(result[key])]
```

I will update code for this along with test for it.

Reply:
Yes, that looks good. In particular, add an idempotency test for this one.